src/liballoc/collections/btree/navigate.rs

```diff
@@ -158,8 +103,9 @@ impl<'a, K, V> Handle<NodeRef<marker::Mut<'a>, K, V, marker::Leaf>, marker::Edge
     /// - The caller must ensure that the leaf edge is not the last one in the tree.
     /// - Using the updated handle may well invalidate the returned references.
     pub unsafe fn next_unchecked(&mut self) -> (&'a mut K, &'a mut V) {
-        let (next_edge, kv) = next_unchecked_mut(ptr::read(self));
-        *self = next_edge;
+        let kv = ptr::read(self).next_kv();
+        let kv = unwrap_unchecked(kv.ok());
+        *self = ptr::read(&kv).next_leaf_edge();
```

Comment:
Please also replace the `*self = ...` with a `ptr::write(self, ...)` because that way it'll do the right thing even if self gains a drop impl or so.